### PR TITLE
support for timespan serialization

### PR DESCRIPTION
**Pull request by julienGrd**
see https://github.com/Blazored/LocalStorage/issues/38

### CODE REVIEW

File: src/Blazored.LocalStorage/LocalStorageService.cs [hunk 1]
```diff
@@ -26,7 +31,7 @@ public async Task SetItemAsync(string key, object data)
             if (e.Cancel)
                 return;
 
-            await _jSRuntime.InvokeAsync<object>("localStorage.setItem", key, JsonSerializer.Serialize(data));
+            await _jSRuntime.InvokeAsync<object>("localStorage.setItem", key, JsonSerializer.Serialize(data, this._jsonOptions));
```
Comment (chrissainty) [inline src/Blazored.LocalStorage/LocalStorageService.cs]: Could we drop the `this` keyword.
```suggestion
            await _jSRuntime.InvokeAsync<object>("localStorage.setItem", key, JsonSerializer.Serialize(data, _jsonOptions));
```

File: src/Blazored.LocalStorage/LocalStorageService.cs [hunk 2]
```diff
@@ -41,7 +46,7 @@ public async Task<T> GetItemAsync<T>(string key)
             if (string.IsNullOrWhiteSpace(serialisedData))
                 return default(T);
 
-            return JsonSerializer.Deserialize<T>(serialisedData);
+            return JsonSerializer.Deserialize<T>(serialisedData, this._jsonOptions);
```
Comment (chrissainty) [inline src/Blazored.LocalStorage/LocalStorageService.cs]: Could we drop the `this` keyword.
```suggestion
            return JsonSerializer.Deserialize<T>(serialisedData, _jsonOptions);
```

File: src/Blazored.LocalStorage/LocalStorageService.cs [hunk 3]
```diff
@@ -73,7 +78,7 @@ public void SetItem(string key, object data)
             if (e.Cancel)
                 return;
 
-            _jSInProcessRuntime.Invoke<object>("localStorage.setItem", key, JsonSerializer.Serialize(data));
+            _jSInProcessRuntime.Invoke<object>("localStorage.setItem", key, JsonSerializer.Serialize(data, this._jsonOptions));
```
Comment (chrissainty) [inline src/Blazored.LocalStorage/LocalStorageService.cs]: Could we drop the `this` keyword.
```suggestion
            _jSInProcessRuntime.Invoke<object>("localStorage.setItem", key, JsonSerializer.Serialize(data, _jsonOptions));
```

File: src/Blazored.LocalStorage/LocalStorageService.cs [hunk 4]
```diff
@@ -91,7 +96,7 @@ public T GetItem<T>(string key)
             if (string.IsNullOrWhiteSpace(serialisedData))
                 return default(T);
 
-            return JsonSerializer.Deserialize<T>(serialisedData);
+            return JsonSerializer.Deserialize<T>(serialisedData, this._jsonOptions);
```
Comment (chrissainty) [inline src/Blazored.LocalStorage/LocalStorageService.cs]: Could we drop the `this` keyword.
```suggestion
            return JsonSerializer.Deserialize<T>(serialisedData, _jsonOptions);
```

**Comment by chrissainty** [comment]
Did you mean to close this PR @julienGrd ? Do you no longer want it merged?

**Comment by julienGrd** [comment]
I was sure i made a mistake with this PR ! sorry i reopen it and merge it !

**Pull request merged**